File: Python/백준/Silver/10866. 덱/덱.py
import sys
from collections import deque
input = sys.stdin.readline

dq = deque()

N = int(input())
for _ in range(N):
    command = list(map(str, input().split()))

    match command[0]:
        case "push_front":
            dq.appendleft(command[1])
        case "push_back":
            dq.append(command[1])
        case "pop_front":
            if len(dq) != 0:
                print(dq.popleft())
            else:
                print(-1)
        case "pop_back":
            if len(dq) != 0:
                print(dq.pop())
            else:
                print(-1)
        case "size":
            print(len(dq))
        case "empty":
            if len(dq) != 0:
                print(0)
            else:
                print(1)
        case "front":
            if len(dq) != 0:
                print(dq[0])
            else:
                print(-1)
        case "back":
            if len(dq) != 0:
                print(dq[-1])
            else:
                print(-1)

File: Java/백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;

        Deque<Integer> dq = new ArrayDeque<>();

        int N = Integer.parseInt(br.readLine());
        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());

            switch (st.nextToken()) {
                case "push_front":
                    dq.offerFirst(Integer.parseInt(st.nextToken()));
                    break;
                case "push_back":
                    dq.offer(Integer.parseInt(st.nextToken()));
                    break;
                case "pop_front":
                    sb.append(dq.isEmpty() ? -1 : dq.poll()).append("\n");
                    break;
                case "pop_back":
                    sb.append(dq.isEmpty() ? -1 : dq.pollLast()).append("\n");
                    break;
                case "size":
                    sb.append(dq.size()).append("\n");
                    break;
                case "empty":
                    sb.append(dq.isEmpty() ? 1 : 0).append("\n");
                    break;
                case "front":
                    sb.append(dq.isEmpty() ? -1 : dq.peek()).append("\n");
                    break;
                case "back":
                    sb.append(dq.isEmpty() ? -1 : dq.peekLast()).append("\n");
                    break;
                default:
                    break;
            }
        }

        System.out.println(sb);

    }
}